src/assisted_test_infra/test_infra/helper_classes/nodes.py

```diff
@@ -26,6 +26,11 @@ def __init__(self, node_controller: NodeController):
         self.controller = node_controller
         self._nodes = None
         self._nodes_as_dict = None
+        self.__is_prepared = False
```

Comment:
since nothing inherits ``Nodes`` I don't see a reason not to have:
```suggestion
        self._is_prepared = False
```

Reply:
I don't want it will be accessed from outside the class, setting private property will raise an exception

Reply:
if we have someone doing ``nodes._<private_field>`` then we have bigger problems to deal with :joy:

I'm not opposing to it in a hard way, but I don't see a difference from a case someone will touch ``self._nodes`` and leave ``self._nodes_as_dict`` untouched. Of course it cannot be a good thing. So some coherence is expected, at least at the same class that we're dealing with